packages/cfd/src/Components/cfd-dxtrade-compare-content.tsx

```diff
@@ -0,0 +1,476 @@
+             */
+            has_reality_check?: 0 | 1;
+            /**
+             * Allowed contract types
+             */
+            legal_allowed_contract_categories?: string[];
+            /**
+             * Allowable currencies
+             */
+            legal_allowed_currencies?: string[];
+            /**
+             * Allowable markets
+             */
+            legal_allowed_markets?: string[];
+            /**
+             * Default account currency
+             */
+            legal_default_currency?: string;
+            /**
+             * Landing Company legal name
+             */
+            name?: string;
+            /**
+             * Legal requirements for the Landing Company
+             */
+            requirements?: {
+                [k: string]: unknown;
+            };
+            /**
+             * Landing Company short code
+             */
+            shortcode?: string;
+            /**
+             * Flag that indicates whether the landing company supports professional accounts or not
+             */
+            support_professional_client?: 0 | 1;
+        };
+    };
+};
+
+type TCFDAttributeDescriberProps = {
+    name: string;
+    counter: number | null;
+};
+
+type TFilterAvailableAccounts = (
+    landing_companies: TLandingCompany,
+    table: TAccountsDescription[],
+    is_logged_in: boolean,
+    show_eu_related: boolean,
+    platform: string,
+    is_australian: boolean
+) => Array<{ [key: string]: string | React.ReactNode | undefined }>;
+
+type TAccountsDescription = {
+    attribute: string;
+    dxtrade: TDxTradeAccountsDescription;
+};
+
+type TDxTradeAccountsDescription = {
+    derivx: string;
+    footnote: string | null;
+};
+
+type TCompareAccountsReusedProps = {
+    landing_companies: TLandingCompany;
+    platform: string;
+    is_logged_in: boolean;
+    is_uk: boolean;
+};
+
+type TCompareAccountsDataParams = TCompareAccountsReusedProps & {
+    show_eu_related: boolean;
+    residence: string;
+};
+
+type TCFDCompareAccountHintProps = TCompareAccountsReusedProps & {
+    show_risk_message: boolean;
+};
+
+type TModalContentProps = TCompareAccountsReusedProps & {
+    show_eu_related: boolean;
+    residence: string;
+    is_eu: boolean;
+};
+
+type TGetAccounts = (params: TCompareAccountsReusedProps) => TAccountsDescription[];
+
+type TAccountTypesToFilter = (TLandingCompany['dxtrade_all_company'] | boolean | undefined)[];
+
+const getAccounts: TGetAccounts = ({ landing_companies, is_logged_in }) => {
+    const getLoggedInTypesCount = () =>
+        ([landing_companies?.dxtrade_all_company] as TAccountTypesToFilter).filter(Boolean).length;
```

Comment:
```suggestion
        ([landing_companies?.dxtrade_all_company] as TAccountTypesToFilter).length;
```
Why `filter(Boolean)` ? Isn't getting the length enough in this case?

Reply:
Sorted.

Reply:
I just use it to make sure that null properties are not included in the condition.